### PR TITLE
MAISTRA-723 use port from NetworkEndpoint on ServiceInstance when con…

### PR DESCRIPTION
**Pull request by rcernich**
…figuring Gateway listeners

Signed-off-by: rcernich <rcernich@redhat.com>

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/gateway.go [hunk 1]
```diff
@@ -59,13 +59,20 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(env *model.Environme
 	errs := &multierror.Error{}
 	listeners := make([]*xdsapi.Listener, 0, len(mergedGateway.Servers))
 	for portNumber, servers := range mergedGateway.Servers {
+		var si *model.ServiceInstance
+		for _, w := range workloadInstances {
+			if w.Endpoint.ServicePort.Port == int(portNumber) {
+				si = w
```
Comment (luksa) [inline pilot/pkg/networking/core/v1alpha3/gateway.go:65]: Missing `break` after this line?

Comment (rcernich) [reply pilot/pkg/networking/core/v1alpha3/gateway.go:65]: Yep, although I think it's innocuous as any matching ServiceInstance will have the same NetworkEndpoint.Port, but certainly less efficient.

Good catch!

**Review by knrc — APPROVED** [review]

**Review by jwendell — APPROVED** [review]

File: pilot/pkg/networking/core/v1alpha3/gateway.go [hunk 2]
```diff
@@ -59,13 +59,21 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(env *model.Environme
...
+		}
+
 		// on a given port, we can either have plain text HTTP servers or
 		// HTTPS/TLS servers with SNI. We cannot have a mix of http and https server on same port.
 		opts := buildListenerOpts{
-			env:        env,
-			proxy:      node,
-			bind:       WildcardAddress,
-			port:       int(portNumber),
+			env:   env,
```
Comment (dmitri-d) [inline pilot/pkg/networking/core/v1alpha3/gateway.go:73]: Nit: probably worth it to preserve the original formatting here to (hopefully) avoid merge conflicts in the future.

**Review by dmitri-d — APPROVED** [review]

**Comment by rcernich** [comment]
@luksa, mergify want's your review.  sorry to bother.

**Review by luksa — APPROVED** [review]

**Pull request merged**